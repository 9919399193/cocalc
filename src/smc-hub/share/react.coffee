###
Load react support for rendering.
###

async = require('async')

# Code for rendering react components to html.
ReactDOMServer = require('react-dom/server')

require('./jsdom-support')

# Uncomment for cc-in-cc dev benchmarking purposes.  This variable is already set
# by the Docker container when running in kubernetes.
## process.NODE_ENV="production"

require('smc-webapp/r_misc').SHARE_SERVER = true

# Load katex jQuery plugin.
require('smc-webapp/jquery-plugins/katex')

exports.react = (res, component, extra, viewer) ->
    res.type('html')
    t0 = new Date()
    stream = ReactDOMServer.renderToStaticNodeStream(component)
    stream.pipe(res)
    stream.once 'end', ->
        console.log("react: time to render and stream out: #{new Date() - t0}ms", extra)
