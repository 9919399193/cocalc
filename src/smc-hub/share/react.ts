/*
Load react support for rendering.
*/

// Code for rendering react components to html.
import * as ReactDOMServer from "react-dom/server";

import "./jsdom-support";

// Uncomment for cc-in-cc dev benchmarking purposes.  This variable is already set
// by the Docker container when running in kubernetes.
//# process.NODE_ENV="production"

import { set_share_server } from "smc-webapp/r_misc/share-server";

set_share_server(true);

// Load katex jQuery plugin.
require("smc-webapp/jquery-plugins/katex");

export function react(res, component, extra: any = ""): void {
  res.type("html");
  const t0 = new Date().valueOf();
  const stream = ReactDOMServer.renderToStaticNodeStream(component);
  stream.pipe(res);
  stream.once("end", () =>
    console.log(
      `react: time to render and stream out: ${new Date().valueOf() - t0}ms`,
      extra
    )
  );
}
