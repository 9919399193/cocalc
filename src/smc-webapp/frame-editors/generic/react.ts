// THIS SHOULD BE MOVED OUT OF frame-editors/


export const {
  React,
  ReactDOM,
  rclass,
  rtypes,
  Fragment,
  redux,
  Redux
} = require("smc-webapp/smc-react");

import * as ReactOrig from "react";

export const Component = ReactOrig.Component;

export type Rendered = ReactOrig.ReactElement<any> | undefined;
