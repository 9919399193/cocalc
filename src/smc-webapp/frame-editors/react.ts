export const {
  React,
  ReactDOM,
  rclass,
  rtypes,
  Fragment,
  redux,
  Redux
} = require("smc-webapp/smc-react");

import * as ReactOrig from "react";

export const Component = ReactOrig.Component;

export type Rendered = ReactOrig.ReactElement<any> | undefined;
