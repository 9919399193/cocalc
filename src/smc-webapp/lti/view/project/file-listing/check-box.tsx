import * as React from "react";
import { assert_never } from "../../../helpers";

export enum Mark {
  check,
  slash,
  empty
}

interface Props {
  fill: Mark;
  on_click: (fill: Mark) => void;
}

export function CheckBox({ fill, on_click }: Props) {
  switch (fill) {
    case Mark.check:
      return <span onClick={() => on_click(fill)}>☑</span>;
    case Mark.slash:
      return <span onClick={() => on_click(fill)}>⧅</span>;
    case Mark.empty:
      return <span onClick={() => on_click(fill)}>☐</span>;
    default:
      return assert_never(fill);
  }
}
