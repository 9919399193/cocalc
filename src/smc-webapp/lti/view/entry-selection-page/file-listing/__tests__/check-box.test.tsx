import * as React from "react";
import { shallow } from "enzyme";
import { CheckBox, Mark } from "../check-box";

describe("Render regressions", () => {
  test("Mark.check", () => {
    const rendered = shallow(<CheckBox fill={Mark.check} on_click={_ => {}} />);

    expect(rendered).toMatchSnapshot("Box is a check mark");
  });

  test("Mark.slash", () => {
    const rendered = shallow(<CheckBox fill={Mark.slash} on_click={_ => {}} />);

    expect(rendered).toMatchSnapshot("Box is a slash");
  });

  test("Mark.empty", () => {
    const rendered = shallow(<CheckBox fill={Mark.empty} on_click={_ => {}} />);

    expect(rendered).toMatchSnapshot("Box is empty");
  });
});

describe("Interactions", () => {
  test("Click", () => {
    const mock_on_click = jest.fn();
    const evt = new MouseEvent("click");
    const fill = Mark.check;
    const rendered = shallow(<CheckBox fill={fill} on_click={mock_on_click} />);

    rendered.simulate("click", evt);
    expect(mock_on_click.mock.calls.length).toBe(1);
    expect(mock_on_click.mock.calls[0][0]).toBe(fill);
  });
});
