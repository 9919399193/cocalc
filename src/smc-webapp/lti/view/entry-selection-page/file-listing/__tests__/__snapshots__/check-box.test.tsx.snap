// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render regressions Mark.check: Box is a check mark 1`] = `
<span
  onClick={[Function]}
>
  ☑
</span>
`;

exports[`Render regressions Mark.empty: Box is empty 1`] = `
<span
  onClick={[Function]}
>
  ☐
</span>
`;

exports[`Render regressions Mark.slash: Box is a slash 1`] = `
<span
  onClick={[Function]}
>
  ⧅
</span>
`;
